pkg/util/interrupt.go

```diff
@@ -0,0 +1,95 @@
+/*
+Derived from:
+
+https://github.com/kubernetes/kubernetes/blob/master/pkg/util/interrupt/interrupt.go
+
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+// childSignals are the allowed signals that can be sent to children in Unix variant OS's
+var childSignals = []os.Signal{syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT}
```

Comment:
We still need to understand the choice of what signals to wait for as I asked earlier.

We're missing SIGHUP. According to the [docs](https://golang.org/pkg/os/signal/#hdr-Default_behavior_of_signals_in_Go_programs) and local experimentation, if the program receives SIGHUP it exits immediately.

That means that if I am running `s2i` and close my terminal window, there will be no clean up.

On SIGQUIT I just want to understand our decision. If we wouldn't wait for SIGQUIT for perform the clean up, we wouldn't need to have the `dumpStack` function and simply let the Go runtime do its default action.
In what scenarios are we expecting a SIGQUIT?